src/main/scala/is/hail/expr/ir/SpecializedArrayBuilders.scala

```diff
@@ -293,4 +384,30 @@ class BooleanArrayBuilder(initialCapacity: Int) extends MissingArrayBuilder(init
     b(i) = x
     missing(i) = false
   }
+
+  def sort(ordering: BooleanOrderingFunction): Unit = {
+    var newend = 0
+    var i = 0
+    while (i < size_) {
+      if (!missing(i)) {
+        if (newend != i) {
+          b(newend) = b(i)
+          missing(newend) = false
+        }
+        newend += 1
+      }
+      i += 1
+    }
+    i = newend
+    while (i < size) {
+      missing(i) = true
+      i += 1
+    }
+    val newb = b.take(newend).sortWith(ordering(_, _))
```

Comment:
I'm pretty sure both these operations will induce boxing, which is kind of unavoidable without writing our own sort, but it feels a bit odd to go out of our way to avoid boxing in the comparison function